evaluator.plandefinition/src/test/resources/org/opencds/cqf/cql/evaluator/plandefinition/r4/anc-dak/data-bundle.json

```diff
@@ -0,0 +1,236 @@
+					"system": "http://example.org/fhir/NamingSystem/identifiers",
+					"value": "anc-b8-de27-example"
+				}],
+				"status": "final",
+				"code": {
+					"coding": [{
+						"system": "http://fhir.org/guides/who/anc-cds/CodeSystem/anc-custom-codes",
+						"code": "ANC.B8.DE27",
+						"display": "Symptoms of severe pre-eclampsia"
+					}]
+				},
+				"subject": {
+					"reference": "Patient/5946f880-b197-400b-9caa-a3c661d23041"
+				},
+				"encounter": {
+					"reference": "Encounter/helloworld-patient-1-encounter-1"
+				},
+				"performer": [{
+					"reference": "PractitionerRole/anc-practitionerrole-example"
+				}],
+				"valueCodeableConcept": {
+					"coding": [{
+						"system": "http://fhir.org/guides/who/anc-cds/CodeSystem/anc-custom-codes",
+						"code": "ANC.B8.DE29",
+						"display": "Vomiting"
+					}]
+				}
+			}
+		},
+		{
+			"resource": {
+				"resourceType": "Observation",
+				"id": "anc-b9-de144-example",
+				"meta": {
+					"profile": ["http://fhir.org/guides/who/anc-cds/StructureDefinition/anc-b9-de144"]
+				},
+				"identifier": [{
+					"use": "official",
+					"system": "http://example.org/fhir/NamingSystem/identifiers",
+					"value": "anc-b9-de144-example"
+				}],
+				"status": "final",
+				"code": {
+					"coding": [{
+						"system": "http://fhir.org/guides/who/anc-cds/CodeSystem/anc-custom-codes",
+						"code": "ANC.B9.DE144",
+						"display": "Urine dipstick result - protein"
+					}]
+				},
+				"subject": {
+					"reference": "Patient/5946f880-b197-400b-9caa-a3c661d23041"
+				},
+				"encounter": {
+					"reference": "Encounter/helloworld-patient-1-encounter-1"
+				},
+				"performer": [{
+					"reference": "PractitionerRole/anc-practitionerrole-example"
+				}],
+				"valueCodeableConcept": {
+					"coding": [{
+						"system": "http://fhir.org/guides/who/anc-cds/CodeSystem/anc-custom-codes",
+						"code": "ANC.B9.DE146",
+						"display": "+"
+					}]
+				}
+			}
+		},
+		{
+			"resource": {
+				"resourceType": "Encounter",
+				"id": "helloworld-patient-1-encounter-1",
+				"meta": {
+					"profile": ["http://hl7.org/fhir/us/qicore/StructureDefinition/qicore-encounter"]
+				},
+				"status": "finished",
+				"class": {
+					"system": "http://terminology.hl7.org/CodeSystem/v3-ActCode",
+					"code": "AMB",
+					"display": "ambulatory"
+				},
+				"type": [{
+					"coding": [{
+						"system": "http://snomed.info/sct",
+						"version": "2020-09",
+						"code": "185463005",
+						"display": "Visit out of hours (procedure)"
+					}]
+				}],
+				"subject": {
+					"reference": "Patient/5946f880-b197-400b-9caa-a3c661d23041"
+				},
+				"period": {
+					"start": "2020-01-01T10:00:00-07:00",
+					"end": "2020-01-01T11:00:00-07:00"
+				}
+			},
+			"request": {
+				"method": "PUT",
+				"url": "Encounter/helloworld-patient-1-encounter-1"
+			}
+		},
+		{
+		     "resource": {
+		        "resourceType": "Patient",
+		        "id": "Patient/5946f880-b197-400b-9caa-a3c661d23041",
```

Comment:
This id is likely the cause of your issues.  Remove the "Patient/" from the id and it should be able to find this patient to use as the context.

Reply:
I have just done and still get a context value of null. In case this is helpful, below are the values of the various parameters in `BundleRetrieveProvider:retrieve`:

context: `Encounter`
contextPath: `encounter`
contextValue: `null`
dataType: `Observation`
templateId: `http://hl7.org/fhir/StructureDefinition/Observation`
codePath: `code`
codes: `null`
valueSet: `http://fhir.org/guides/who/anc-cds/ValueSet/anc-b8-de17`
datePath: `null`
dateLowPath: `null`
dateHighPath: `null`
dateRange: `null`

Reply:
Apologies for the delays here, looking at this briefly, I notice that the "context" is "Encounter", is there a reason you are choosing an Encounter context rather than a Patient context?

Reply:
I do not call this method directly. It is the CQL engine/framework that calls it: https://github.com/cqframework/clinical-reasoning/blob/v2.6.0/evaluator.fhir/src/main/java/org/opencds/cqf/cql/evaluator/engine/retrieve/BundleRetrieveProvider.java#L50-L55

Reply:
Right, but that's already after the engine is running with the parameters that it has been set up with. This would be earlier in the process where you are actually setting up the engine context.

Reply:
Looking at the test I don't see anything that would set it up for Encounter context. We have an engineer looking at this now and will get back to you as soon as possible. Again, apologies for the delays here.

Reply:
Thank you so much for looking at it. I am anxiously looking forward to the response! 😊

Reply:
@dkayiwa , I'm able to successfully run the ANCDT17 test. Is that the expected result, or is the test failing for you?

Reply:
@brynrhodes  that is another thing that i wanted bring up after this has been sorted. The test passes regardless of the patient observation values. In other words, the output care plan is always exactly the same whether the CQL evaluations return true or false.

Reply:
I see, so there's something more fundamental going on. We'll keep digging.

Reply:
Thank you so much @brynrhodes 👍

Reply:
@dkayiwa , we think we found and addressed an issue with this, in that the PlanDefinition $apply processor wasn't creating actions if the PlanDefinition.action didn't have a definition or any child actions. That fix has been merged to master (https://github.com/cqframework/clinical-reasoning/pull/266) However, I'm not sure we ever reproduced the null context though, so I'd like to see if you can recreate the behavior with the latest from master if that's alright?

Reply:
We pulled the tests from this PR into that one, so you should be able to just check and see whether you're getting expected behavior.